R/functions_netCDF.R

```diff
@@ -456,42 +687,377 @@ create_netCDF_from_array_with_variables <- function(x, locations, grid,
+#'  longitude_of_prime_meridian = 0.0,
+#'  semi_major_axis = 6378137.0,
+#'  inverse_flattening = 298.257223563,
+#'  crs_wkt = 'GEOGCS["WGS 84",DATUM["WGS_1984",SPHEROID["WGS 84",6378137,298.257223563,AUTHORITY["EPSG","7030"]],
+#'                         AUTHORITY["EPSG","6326"]],PRIMEM["Greenwich",0, AUTHORITY["EPSG","8901"]],
+#'                         UNIT["degree",0.01745329251994328,AUTHORITY["EPSG","9122"]],AUTHORITY["EPSG","4326"]]'
+#' )
+#'
+#' # global attributes
+#' global_attributes <- list(
+#'    title = "",
+#'    institution = 'Southwest Biological Science Center, U.S. Geological Survey',
+#'    description = 'how this data was made',
+#'    source = paste(
+#'      "SOILWAT2 (v4.2.0);",
+#'      "rSOILWAT2 (v2.3.2);",
+#'      "rSFSW2 (v3.1.2)"
+#'    ),
+#'    source_id = "SOILWAT2",
+#'    realm = "land",
+#'    parent_mip_era = "CMIP5",
+#'    parent_experiment_id = "RCP45",
+#'    parent_source = "CanESM2",
+#'    parent_variant_label = "r1i1p1f1",
+#'    product = "model-output",
+#'    projection = 'Geographic',
+#'    grid = 'WGS84',
+#'    grid_label = "gn",
+#'    nominal_resolution = "10 km", # calculate_nominal_resolution
+#'    further_info_url = "https://github.com/DrylandEcology/",
+#'    contact = "you@email.com"
+#'    )
+#'
+#' # run create_empty_netCDF_file function --------------------------------------
+#' create_empty_netCDF_file(
+#'      data = data,
+#'      has_T_timeAxis = TRUE,
+#'      has_Z_verticalAxis = FALSE,
+#'      time_bounds = annual_time_bounds,
+#'      vert_bounds = NULL,
+#'      var_attributes = var_attributes,
+#'      time_attributes  = time_attributes,
+#'      vertical_attributes = NULL,
+#'      global_attributes = global_attributes,
+#'      crs_attributes = crs_attributes,
+#'      isGridded = TRUE,
+#'      locations = locations,
+#'      file = outFileName,
+#'      force_v4 = TRUE,
+#'      overwrite = TRUE
+#'    )
+#'
+#' # run function populate_netcdf_from_array
+#' populate_netcdf_from_array(
+#'      file = outFileName,
+#'      data = data,
+#'      var_names = var_attributes$name,
+#'      has_T_timeAxis = TRUE,
+#'      has_Z_verticalAxis = FALSE,
+#'      isGridded = TRUE,
+#'      locations = locations,
+#'      force_v4 = TRUE
+#'    )
+#'    
+#'  unlink(outFileName)    
+#'
+#' @seealso \code{create_empty_netCDF_file}
+#' @seealso \url{http://cfconventions.org/cf-conventions/cf-conventions.html} # for defining attributes
+#' @seealso Demonstration of netCDF functionality in rSW2analysis vignette
+#'
+#' @export
 
-  #--- prepare to put values to full grid format
-  val_template <- rep(NA, raster::ncell(grid))
-  val_ids <- raster::cellFromXY(grid, locations)
+populate_netcdf_from_array <- function(file, data, var_names = NULL,
+                                       has_T_timeAxis, has_Z_verticalAxis,
+                                       isGridded = TRUE, grid = NULL, locations,
+                                       force_v4 = TRUE, verbose = FALSE) {
```

Comment:
Function `populate_netcdf_from_array` does not use the argument `force_v4` --> remove